### PR TITLE
Add data member accessor expression

### PR DESCRIPTION
**Pull request by roberteg16**

### CODE REVIEW

File: tmplang/include/tmplang/Tree/Source/Exprs.h [hunk 1]
```diff
@@ -124,6 +126,50 @@ class ExprVarRef final : public Expr {
...
+  tmplang::SourceLocation getEndLoc() const override {
+    return Field.getSpan().End;
+  }
+
+  static bool classof(const Node *node) {
+    return node->getKind() == Kind::ExprDataFieldAccess;
+  }
+
+private:
+  BaseType Base;
```
Comment (d-alonso) [inline tmplang/include/tmplang/Tree/Source/Exprs.h:168]: Naming it BaseType is a bit confusing. It's just the type of the base node, so AccessedNode or BaseNode seems more appropriate. This way it seems like you're describing it as a "type" itself (int, struct etc)

**Review by d-alonso — COMMENTED** [review]
Stellar workkk

File: tmplang/include/tmplang/Tree/Source/Exprs.h (hunk 1 above)
Comment (roberteg16) [reply tmplang/include/tmplang/Tree/Source/Exprs.h:168]: Totally agree, I'll address this on a new PR

Comment (roberteg16) [reply tmplang/include/tmplang/Tree/Source/Exprs.h:168]: https://github.com/tmplang/tmplang/pull/54

**Pull request merged**